src/modules/order/order.service.ts

```diff
@@ -44,6 +57,11 @@ export async function updateOrder(
 ) {
   const itemsTotal = await getItemTotal(server, input);
   const servicesTotal = await getServiceTotal(server, input);
+  const tipAmount =
+    input.tip?.type === TipType.percentage
+      ? ((input.tip?.amount ?? 0) / 100) *
+        (itemsTotal.price + servicesTotal.price)
+      : input.tip?.amount ?? 0;
```

Comment:
```suggestion
  const tipAmount = !input.tip ? 0 : (input.tip.type === TipType.percentage......)
```

This would require less null coalescing operators chaining